src/libponyc/pkg/buildflagset.c

```diff
@@ -534,6 +534,35 @@ bool define_build_flag(const char* name)
 }
 
 
+bool remove_build_flags(const char* flags[])
+{
+  pony_assert(flags != NULL);
+
+  if(_user_flags == NULL)
+  {
```

Comment:
It seems as though you should be able to `return false;` here if `_user_flags == NULL` since that means no user flags have been defined.

Reply:
Yea, I actually coded it that way, but then thought its possible the initialization could potentially add flags and thought it best not to assume anything and just fall through.

Reply:
No, initialization doesn't add any flags. All it does is initialize the hashmap. See: https://github.com/ponylang/ponyc/blob/master/src/libponyrt/ds/hash.c#L180-L202 for what `flagtab_init` does.

Note: I should clarify that the way the hashmap implementation works there's a bunch of template macros which effectively alias the hashmap functions for a given type/name. If you find a `DECLARE_HASHMAP(flagtab, flagtab_t, flag_t);` with the same prefix then that is what is happening and you can look up the relevant hashmap functions to see what is happening if you need to.

Reply:
I absolutely agree that now we could return early, but I see no downside to letting the normal processing proceed. I also tend to like code that has only and one exit point.

Reply:
Okey, in that case you can change this logic to be `if(_user_flags != NULL) { for... }` and keep a single exit point. Or you can keep things as they are. Like the other item, I don't feel strongly about it........

Reply:
Then unless overriden by higher powers, I'll leave it as is :)